### ISSUE TITLE
Source code mounts not using cached consistency, Yarn extremely slow

### ISSUE THREAD

**Issue by miketamis** [microsoft/vscode-remote-release#879]
When running yarn install inside the default node (LTS) container, it takes a long time to run.
```
[4/4] Building fresh packages...
success Saved lockfile.
Done in 625.55s.
```
Using macOS Mojave

Using Docker Desktop
Version 2.0.0.3 (31259)
Engine: 18.09.2

**Comment by sandorfr** [comment]
I noticed that too, using [cached consistency](https://code.visualstudio.com/docs/remote/containers-advanced#_changing-the-default-source-code-mount) helped not sure it resolved everything though

**Comment by egamma** [comment]
@Chuxel this would be a good addition to the tips and tricks section. The other way to speed this up is to create a volume and then do development inside the volume.

**Comment by Chuxel** [comment]
@egamma The is one of the reasons we added `workspaceMount`. Unfortunately it is waiting on #442 for a full fix since you need to use an absolute path.  I also wonder if we shouldn't use cached for the default mount.

**Comment by Chuxel** [comment]
@egamma I forgot to mention that the cached consistency link above is from our docs... I could add something into tips and tricks that references it, but I think we should consider these +1s for #442 as well along with considering using cached by default.

**Comment by sandorfr** [comment]
> @egamma I forgot to mention that the cached consistency link above is from our docs... I could add something into tips and tricks that references it, but I think we should consider these +1s for #442 as well along with considering using cached by default.

yes I guess making it the default would save you from a lot of requests from people not reading the docs. Also as soon as a project is large enough it really the only viable option. At the moment it makes using docker compose pretty mandatory.

I would fix #442 with high priority though, docker might improve options on various operating system (maybe a tmpfs option will one day surface on windows, just guessing). You don't want people to be unable to harness future docker improvements by locking things done too much.

**Comment by Chuxel** [comment]
The specifics of this issue was actually resolved a while ago. There's the ability to pick the mount consistency as a property along with tips for using volume mounts to speed things up or using Open Repo in Container which uses a volume. Thanks!
